### ISSUE TITLE
Rule Request: Prefer CGRect over NSRect

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#2371]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

CGRect and NSRect are the same. NSRect is just a `typealias` of CGRect. I would like to enforce CGRect for consistency with iOS code.

This would apply to:

- `NSRect` => `CGRect`
- `NSSize` => `CGSize`
- `NSPoint` => `CGPoint`

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Would trigger:

```swift
NSSize(width: 100, height: 100)

let foo: NSSize = bar()
````

Would not trigger:

```swift
CGSize(width: 100, height: 100)

let foo: CGSize = bar()
```

3. Should the rule be configurable, if so what parameters should be configurable?

No, there's nothing to configure.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.

I think it should be enabled by default as there's really no good reason to use the `NS` versions. They're just type aliases to the CG versions and most people use them inconsistently anyway.

**Comment by rpendleton** [comment]
I'm not sure what SwiftLint requires before making a rule part of the included set of rules, but depending on how many others want something like this, it seems like this would be simple enough to implement as a custom_rule instead:

```
custom_rules:
  no_nsrect:
    name: "Use CGRect instead of NSRect"
    regex: "NSRect"
  no_nssize:
    name: "Use CGSize instead of NSSize"
    regex: "NSSize"
  no_nspoint:
    name: "Use CGPoint instead of NSPoint"
    regex: "NSPoint"
```
